src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
     firstMount: true,
   };
 
-  componentWillReceiveProps() {
+  componentDidMount() {
+    if (this.props.variant === 'temporary') {
+      this.enableSwipeHandling();
+    }
+    this.mounted = true;
+  }
+
+  componentWillReceiveProps(nextProps) {
     this.setState({
       firstMount: false,
     });
+
+    if (this.props.variant !== 'temporary' && nextProps.variant === 'temporary') {
+      this.enableSwipeHandling();
+    } else if (this.props.variant === 'temporary' && nextProps.variant !== 'temporary') {
+      this.disableSwipeHandling();
+    }
+  }
+
+  componentWillUnmount() {
+    this.disableSwipeHandling();
+    this.removeBodyTouchListeners();
+    this.mounted = false;
+  }
+
+  onBodyTouchStart = event => {
+    // sometimes the event handler is called after unbinding it
```

Comment:
I have never experienced this problem with react-swipeable-views. I know it can happen in IE11, but it's not a touch browser.

Reply:
Re-checked that, it's not needed…